R/tmb-methods.R

```diff
@@ -50,12 +50,34 @@ fitted.mmrm_tmb <- function(object, ...) {
 #' model.frame(object, full = TRUE)
 model.frame.mmrm_tmb <- function(formula, full = FALSE, ...) {
   assert_flag(full)
-  if (full) {
+  dots <- list(...)
+  if (!identical(dots$na.action %||% getOption("na.action"), "na.omit")) {
+    warning("na.action is always set to `na.omit` for `mmrm`!")
+  }
+  if (is.null(dots$data)) {
+    if (full) {
+      formula$tmb_data$full_frame
```

Comment:
here we can further simplify the logic: if data is not provided, we always return `formula$tmb_data$full_frame`, with selected cols. If `full`, we provide all columns. If not `full`, we remove visits and usubjids.